MVC-DataExploration/Controllers/ProductController.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using System.Web.Mvc;
+
+namespace MVC_DataExploration.Controllers
+{
+    public class ProductController : Controller
+    {
+        // GET: ProductController
+        public ActionResult Index(Models.Product mp)
```

Comment:
You are close; think about where the data for the Index view is coming from in this case - the database.
So you don't need to pass anything into the Index method